File: MFDLabs.Grid.Bot/Commands/GetJobDiagnostics.cs
using System.Linq;
using System.Threading.Tasks;
using Discord.WebSocket;
using MFDLabs.Grid.Bot.Extensions;
using MFDLabs.Grid.Bot.Interfaces;
using MFDLabs.Grid.Bot.Utility;

namespace MFDLabs.Grid.Bot.Commands
{
    internal class GetJobDiagnostics : IStateSpecificCommandHandler
    {
        public string CommandName => "Get Grid Server Job Diagnostics";
        public string CommandDescription => $"Attempts to call a DiagEx SOAP action via the SoapUtility\nLayout: {Settings.Singleton.Prefix}jobdiagnostics jobID type?=1.";
        public string[] CommandAliases => new string[] { "jd", "jobdiag", "jobdiagnostics" };
        public bool Internal => true;
        public bool IsEnabled { get; set; } = true;

        public async Task Invoke(string[] messageContentArray, SocketMessage message, string originalCommand)
        {
            if (!await message.RejectIfNotAdminAsync()) return;

            var jobID = messageContentArray.ElementAtOrDefault(0);
            if (!int.TryParse(messageContentArray.ElementAtOrDefault(1), out int type)) type = 1;

            if (jobID == default)
            {
                await message.ReplyAsync($"Missing required parameter 'jobId', the layout is: {Settings.Singleton.Prefix}{originalCommand} jobID type?=1");
                return;
            }

            await message.ReplyAsync(LuaUtility.Singleton.ParseLuaValues(await SoapUtility.Singleton.DiagExAsync(type, jobID)));
        }
    }
}

File: Source/Lib/Commands/GetJobDiagnostics.ts
import { Message } from 'discord.js';
import { GlobalConfig } from '../Utility/GlobalConfig';
import { AdminUtil } from '../Administration/AdminUtil';
import { ConversionUtil } from '../Utility/ConversionUtil';
import { ScriptUtil } from '../RCCService/ScriptUtility';
import { SOAPUtil } from '../RCCService/SOAPUtility';
import { StateSpecificCommandHandler } from '../CommandUtility/Types/StateSpecificCommandHandler';

class GetJobDiagnostics implements StateSpecificCommandHandler {
	public CommandName = 'Get Job Diagnostcs';
	public CommandDescription = 'Gets diagnostics for a job.';
	public Internal = true;
	public Command = ['jd', 'jobdiag', 'jobdiagnostics'];
	public Callback = async (messageContent: string[], messageRoot: Message) => {
		if (!AdminUtil.RejectIfNotAdmin(messageRoot)) return;

		const jobId = messageContent[0];
		const type = ConversionUtil.ToInt32(messageContent[1]) || 1;

		if (!jobId)
			return messageRoot.reply(
				`Missing required parameter 'jobId', the layout is: ${GlobalConfig.Prefix}jobdiagnostics jobID type?=1`,
			);

		return ScriptUtil.ParseLuaValuesAndRespond(await SOAPUtil.Diag(type, jobId), messageRoot);
	};
}

export = new GetJobDiagnostics();
